module/zfs/spa.c

```diff
@@ -1894,8 +1894,14 @@ spa_load_l2cache(spa_t *spa)
 
 			(void) vdev_validate_aux(vd);
 
-			if (!vdev_is_dead(vd))
-				l2arc_add_vdev(spa, vd);
+			if (!vdev_is_dead(vd)) {
+				boolean_t do_rebuild = B_FALSE;
+
+				(void) nvlist_lookup_boolean_value(l2cache[i],
+				    ZPOOL_CONFIG_L2CACHE_PERSISTENT,
+				    &do_rebuild);
+				l2arc_add_vdev(spa, vd, do_rebuild);
```

Comment:
These lines check if the vdev_label has the attribute ZPOOL_CONFIG_L2CACHE_PERSISTENT. Right now we check both the vdev_label and the header before deciding to rebuild the L2ARC. Technically we could rely only on the header (which contains the necessary information to start the rebuild).
@behlendorf If we rely only on the header, and remove this check here, we could also trigger the L2ARC rebuild in case of `zpool add test cache b.img`.

Reply:
On second thought, we have device `zpool offline` and `online` for this purpose and should not probably trigger L2ARC rebuild upon `zpool add`.

Reply:
We do not need ZPOOL_CONFIG_L2CACHE_PERSISTENT, or do_rebuild. We can rely solely on the header being valid. I will remove them both.